File: Linked List/Arrays/26. Remove Duplicates from Sorted Array/fighting.cpp
#include <vector>
#include <iostream>
using namespace std;

int removeDuplicates(vector<int> &nums)
{
    if(nums.empty()){
        return 0;
    }

    int place = 1;
    for (int i = 1; i < nums.size(); i++)
    {
        if (nums[i] != nums[place - 1])
        {
            nums[place] = nums[i];
            place++;
        }
    }
    return place;
}

int main()
{
    vector<int> nums;
    nums.push_back(1);
    nums.push_back(1);
    nums.push_back(2);
    int x = removeDuplicates(nums);
    cout << x;
    cout << "\n";
    for (int i = 0; i < nums.size(); i++)
    {
        cout << nums[i];
        cout << " ";
    };
}

File: Linked List/Arrays/26. Remove Duplicates from Sorted Array/fighting.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        if not nums:
            return None

        place = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                nums[place] = nums[i]
                place += 1
        
        return place

solution = Solution()
nums = [1, 1, 2]
print(solution.removeDuplicates(nums))
print(nums)
